internals/overlord/servstate/manager_test.go

```diff
@@ -1237,9 +1019,13 @@ checks:
          threshold: 1
          exec:
              command: will-fail
-`, tempFile, s.insertDoneCheck(c, "test2")))
-	err := s.manager.AppendLayer(layer)
-	c.Assert(err, IsNil)
+`
+	s.planAddLayer(c, fmt.Sprintf(
+		layer,
+		tempFile,
+	))
+	s.planChanged(c)
+	checkMgr.PlanChanged(s.plan)
```

Comment:
Why is this manual call to `CheckManager.PlanChanged` needed now? (Here and elsewhere.)

Reply:
This is because the service manager is no longer also a plan manager, and so the action that would give the check manager a plan would be the plan manager calling ```CheckManager.PlanChanged```. These calls are emulating the plan manager receiving the plan update, and propagating this to the service manager and check manager. Does this sound ok ?